compiler/rustc_codegen_llvm/src/consts.rs

```diff
@@ -229,9 +230,17 @@ impl<'ll> CodegenCx<'ll, '_> {
     pub(crate) fn get_static(&self, def_id: DefId) -> &'ll Value {
         let instance = Instance::mono(self.tcx, def_id);
         trace!(?instance);
-        let ty = instance.ty(self.tcx, ty::ParamEnv::reveal_all());
-        trace!(?ty);
-        let llty = self.layout_of(ty).llvm_type(self);
+
+        let DefKind::Static { nested, .. } = self.tcx.def_kind(def_id) else { bug!() };
+        // Nested statics do not have a type, so pick a random type and let `define_static` figure out
```

Comment:
Again, "use a dummy type".

Reply:
I am surprised that we can get away with just making up a random type here that doesn't even have the right size... how does `define_static` figure that out? Where even is that function, I can't find it in the tree nor in this PR.